### PR TITLE
[GraphBolt] Improving `subgraph_sampler` tests.

### PR DESCRIPTION
**Pull request by drivanov**
## Description

The main goal of this PR is to resolve the following 13 warnings that appear when running tests from `test_subgraph_sampler.py`:
```
tests/python/pytorch/graphbolt/test_subgraph_sampler.py: 13 warnings
  /usr/local/lib/python3.10/dist-packages/dgl/graphbolt/item_sampler.py:67: DGLWarning: Unknown item name 'timestamp' is detected and added into `MiniBatch`. You probably need to provide a customized `MiniBatcher`.
    dgl_warning(
```
Some improvements have also been made to eliminate code duplication.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

File: tests/python/pytorch/graphbolt/test_subgraph_sampler.py [hunk 1]
```diff
@@ -884,7 +884,9 @@ def test_SubgraphSampler_Hetero_multifanout_per_layer(sampler_type):
     items_n1 = torch.tensor([0])
     items_n2 = torch.tensor([1])
     names = "seed_nodes"
+    i_len = 2
```
Comment (frozenbugs) [inline tests/python/pytorch/graphbolt/test_subgraph_sampler.py:887]:
```suggestion
    item_length = 2
```

File: tests/python/pytorch/graphbolt/test_subgraph_sampler.py [hunk 2]
```diff
@@ -717,21 +759,9 @@ def test_SubgraphSampler_unique_csc_format_Homo_cpu(labor):
...
-            assert torch.equal(
-                sampled_subgraph.sampled_csc.indices, compacted_indices[step]
-            )
-            assert torch.equal(
-                sampled_subgraph.sampled_csc.indptr, indptr[step]
-            )
-            assert torch.equal(
-                sampled_subgraph.original_column_node_ids, seeds[step]
-            )
+    _datapipe_test_B(
```
Comment (frozenbugs) [inline tests/python/pytorch/graphbolt/test_subgraph_sampler.py:762]: Unnecessary pack, it is fine to have the same group of asserts in each test case.

**Review by frozenbugs — COMMENTED** [review]
LGTM for the warning suppress

File: tests/python/pytorch/graphbolt/test_subgraph_sampler.py (hunk 1 above)
Comment (drivanov) [reply tests/python/pytorch/graphbolt/test_subgraph_sampler.py]: DONE

**Comment by frozenbugs** [comment]
@dgl-bot

**Comment by Rhett-Ying** [comment]
@dgl-bot

File: tests/python/pytorch/graphbolt/test_subgraph_sampler.py [hunk 3]
```diff
@@ -570,18 +574,68 @@ def test_SubgraphSampler_without_dedpulication_Homo_seed_nodes(sampler_type):
...
+                assert torch.equal(
+                    sampled_subgraph.sampled_csc.indptr, indptr[step]
+                )
+                assert torch.equal(
+                    torch.sort(sampled_subgraph.original_column_node_ids)[0],
+                    seeds[step],
+                )
+
+
+def _datapipe_test_A(
```
Comment (frozenbugs) [inline tests/python/pytorch/graphbolt/test_subgraph_sampler.py:597]:
```suggestion
def _assert_hetero_values(
```

File: tests/python/pytorch/graphbolt/test_subgraph_sampler.py [hunk 4]
```diff
@@ -570,18 +574,68 @@ def test_SubgraphSampler_without_dedpulication_Homo_seed_nodes(sampler_type):
...
+                    sampled_subgraph.sampled_csc[etype].indices,
+                    csc_formats[step][etype].indices.to(F.ctx()),
+                )
+                assert torch.equal(
+                    sampled_subgraph.sampled_csc[etype].indptr,
+                    csc_formats[step][etype].indptr.to(F.ctx()),
+                )
+
+
+def _datapipe_test_B(
```
Comment (frozenbugs) [inline tests/python/pytorch/graphbolt/test_subgraph_sampler.py:622]:
```suggestion
def _assert_homo_values(
```

**Review by frozenbugs — APPROVED** [review]

File: tests/python/pytorch/graphbolt/test_subgraph_sampler.py (hunk 3 above)
Comment (drivanov) [reply tests/python/pytorch/graphbolt/test_subgraph_sampler.py]: DONE

**Pull request merged**